### PR TITLE
[#5][Chore] Set up CI/CD workflow for distributing iOS staging app to Firebase

### LINKED ISSUE #5: [Chore] Set up CI/CD workflow for distributing iOS staging app to Firebase
# Why
To release an iOS staging app on Firebase for testing

# Acceptance Criteria
- A Firebase release workflow will be triggered when a PR gets merged into the `develop` branch.
- The CI/CD machine can upload .ipa and .dsyms files to Firebase.

### PR DESCRIPTION
**Pull request by phongvhd93**
- Close #5 

## What happened 👀

- Create a workflow `release_ios_staging` for deploying an iOS staging app to Firebase App Distribution
- Add `.secrets` to `gitignore`

## Insight 📝

When a developer merge a PR to develop branch, GitHub Action will automatically trigger a workflow to deploy the iOS staging app to Firebase App Distribution

## Proof Of Work 📹

TBA

### CODE REVIEW

File: .github/workflows/release_ios_staging.yml [hunk 1]
```diff
@@ -0,0 +1,114 @@
+name: Release iOS Staging App to Firebase
+
+on:
+  push:
+    branches: [develop, chore/#5-set-up-release-firebase-staging-workflow]
```
Comment (suho) [inline .github/workflows/release_ios_staging.yml]: Please remove `chore/#5-set-up-release-firebase-staging-workflow`

File: .github/workflows/release_ios_staging.yml [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+
+
```
Comment (suho) [inline .github/workflows/release_ios_staging.yml:10]: Please one empty line here

File: .github/workflows/release_ios_staging.yml [hunk 3]
```diff
@@ -0,0 +1,114 @@
...
+  ios_release_staging:
+    name: Automatic Release iOS Staging App to Firebase
+    runs-on: macos-latest
+    defaults:
+        run:
+          working-directory: iosApp
+    steps:
+      - name: Cancel Previous Runs
+        uses: styfle/cancel-workflow-action@0.11.0
```
Comment (suho) [inline .github/workflows/release_ios_staging.yml]: You could consider to use the latest approach to cancel previous runs

File: .github/workflows/release_ios_staging.yml [hunk 4]
```diff
@@ -0,0 +1,114 @@
...
+      - name: Bundle install
+        run: bundle install --path vendor/bundle
+
+      - name: Paste Google-Info.plist from .secrets
+        env:
+          IOS_GOOGLE_SERVICE_INFO_STAGING: ${{ secrets.IOS_STAGING_GOOGLE_INFO_PLIST }}
+        run: |
+          mkdir -p PhongKMMIC/Configurations/Plists/GoogleService/Staging
+          touch PhongKMMIC/Configurations/Plists/GoogleService/Staging/GoogleService-Info.plist
+          echo $IOS_GOOGLE_SERVICE_INFO_STAGING | base64 --decode > ./PhongKMMIC/Configurations/Plists/GoogleService/Staging/GoogleService-Info.plist
```
Comment (suho) [inline .github/workflows/release_ios_staging.yml]: I think we can improve this without touch (please validate this)
```suggestion
          echo $IOS_GOOGLE_SERVICE_INFO_STAGING | base64 --decode >> ./PhongKMMIC/Configurations/Plists/GoogleService/Staging/GoogleService-Info.plist
```

File: .gitignore [hunk 5]
```diff
@@ -125,3 +125,4 @@ buildSrc/src/main/kotlin/appPackage/BuildKonfig.kt
 
 # unless supporting rvm < 1.11.0 or doing something fancy, ignore this:
 .rvmrc
+.secrets
```
Comment (suho) [inline .gitignore:128]: Why we need this?

**Review by suho — CHANGES_REQUESTED** [review]

File: .gitignore (hunk 5 above)
Comment (phongvhd93) [reply .gitignore:128]: I kept some secrets to test the workflow locally, so I add the `.secrets` file to the gitignore

**Review by suho — APPROVED** [review]

**Review by markgravity — APPROVED** [review]

**Pull request merged**